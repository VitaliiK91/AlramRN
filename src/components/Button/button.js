import React from 'react';
import {
	TouchableOpacity,
	View,
	StyleSheet,
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import PropTypes from 'prop-types';

const Button = props => (
	<TouchableOpacity
		style={[styles.mainContainer, props.style]}
		onPress={props.onPress}
	>
		{props.icon &&
			<Icon name={props.icon} size={props.size} />
		}
	</TouchableOpacity>
);

const styles = StyleSheet.create({
	mainContainer: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center',
	},
});

Button.defaultProps = {
	onPress: null,
	style: null,
};

Button.propTypes = {
	icon: PropTypes.string.isRequired, // TODO: why string? Mb component?
	size: PropTypes.number.isRequired,
	onPress: PropTypes.func,
	style: View.propTypes.style,
};

export default Button;
