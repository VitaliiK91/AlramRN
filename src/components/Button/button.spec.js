import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './button';

describe('button component', () => {
	it('should render without required props', () => {
		const button = renderer.create(
			<Button
				icon="ios-add"
				size={100}
			/>
		);
		let tree = button.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render with optional props', () => {
		const button = renderer.create(
			<Button
				icon="ios-add"
				size={100}
				onPress={() => 'cat'}
				style={{ flex: 1 }}
			/>
		);
		let tree = button.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should dispatch passed onPress', () => {
		const onPress = jest.fn();
		const button = shallow(
			<Button
				icon="ios-add"
				size={100}
				onPress={onPress}
				style={{ flex: 1 }}
			/>
		);
		button.find('TouchableOpacity').simulate('press');
		expect(onPress).toHaveBeenCalled();
	})
});