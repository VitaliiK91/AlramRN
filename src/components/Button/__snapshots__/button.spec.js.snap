// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component should render with optional props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 100,
        },
        undefined,
        Object {
          "fontFamily": "ExponentFont-mock-ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ï„‚
  </Text>
</View>
`;

exports[`button component should render without required props 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  />
</View>
`;
